content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+  } else {
+    console.log(eventsPage.items.length + ' presence events received in first page');
+    if(eventsPage.hasNext()) {
+      eventsPage.next(function(err, nextPage) { ... });
+    }
+});
+
+bc[ruby]. events_page = channel.presence.history
+puts "#{events_page.items.length} presence events received in first page"
+if events_page.has_next?
+  next_page = events_page.next
+  puts "#{next_page.items.length} presence events received on second page"
+end
+
+bc[php]. $eventsPage = $channel->presence->history();
+echo(count($eventsPage->items) . ' presence events received in first page');
+if ($eventsPage.hasNext()) {
+  $nextPage = $eventsPage->next();
+  echo(count($nextPage->items) . ' presence events received in second page');
+}
+
+```[python]
+events_page = channel.presence.history()
+print str(len(events_page.items)) + " presence events received"
+if events_page.has_next():
+  next_page = events_page.next()
+```
+
+```[java]
+PaginatedResult<PresenceMessage> eventsPage = channel.presence.history(null);
+System.out.println(eventsPage.items().length + " presence events received in first page");
+if(eventsPage.hasNext()) {
+  PaginatedResult<PresenceMessage> nextPage = eventsPage.next();
+  System.out.println(nextPage.items().length + " presence events received in 2nd page");
+}
+```
+
+```[csharp]
+  PaginatedResult<PresenceMessage> eventsPage = await channel.Presence.HistoryAsync();
+  Console.WriteLine(eventsPage.Items.Count + " presence events received in first page");
+  if (eventsPage.HasNext)
+  {
+    PaginatedResult<PresenceMessage> nextPage = await eventsPage.NextAsync();
+    Console.WriteLine(nextPage.Items.Count + " presence events received in 2nd page");
+  }
+```
+
+```[objc]
+[channel.presence history:^(ARTPaginatedResult<ARTPresenceMessage *> *eventsPage, ARTErrorInfo *error) {
+  NSLog(@"%lu presence events received in first page", [eventsPage.items count]);
+  if (eventsPage.hasNext) {
+    [eventsPage next:^(ARTPaginatedResult<ARTPresenceMessage *> *nextPage, ARTErrorInfo *error) {
+      NSLog(@"%lu presence events received in 2nd page", [nextPage.items count]);
+    }];
+  }
+}];
+```
+
+```[swift]
+channel.presence.history { eventsPage, error in
+  let eventsPage = eventsPage!
+  print("\(eventsPage.items.count) presence events received in first page")
+  if eventsPage.hasNext {
+    eventsPage.next { nextPage, error in
+      print("\(nextPage!.items.count) presence events received in 2nd page")
+    }
+  }
+}
+```
+
+```[go]
+page, err := channel.Presence.History(nil)
+fmt.Println("%d messages in first page\n", len(page.PresenceMessages()))
+if page.hasNext() {
+  page2, err := page.Next()
+  fmt.Println("%d messages on 2nd page!\n", len(page2.PresenceMessages()))
+}
+```
+
+See the "presence history documentation":/api/rest-sdk/history#presence-history for further details of the supported query parameters.
+
+h3(#presence-multiple-client-id). Managing multiple client IDs
```

Comment:
This section is in the Miro, but does not come after the "Presence history" section, it should be located in the "Trigger presence events" section.

Reply:
We should also add some explanatory text here - what is "managing multiple client IDs", and why do I need to know about it?